### DISCUSSION TITLE
Issue with multiple PlotViews

### DISCUSSION THREAD

**Discussion by seurrep** [oxyplot/oxyplot#1775] [Q&A]
Currently I'm working with an application that can create documents that can be one of two possible types. The UI for each is different but both display PlotViews as part of their UI and both rely on the same underlying collections to define a set of PlotModels. It's a WPF application using C# intended for desktop use.

Creating a document of one type works fine. The PlotView is displayed. There are no issues.

Problems arise however when creating a document of the second type after having created a document of the first type (or vice versa). I get the dreaded error telling me the model is already being used by some other view. At this point the plotview I'm assuming it's referring to is irrelevant. It's no longer needed & not even being shown (the application itself is using Caliburn Micro and the plotview in question is stored within a screen inside a conductor - note that the unwanted screen is being closed in the call to DeactivateItem() when setting which screen is visible).

Is there any way to forcibly break the association between model and view when it comes to an unwanted plotview? I've tried InvalidatePlot() but this doesn't do the job (I'm guessing because this is more related to refreshing a plotview as opposed to destroying it - the documentation is unclear as to what it does)

**Comment by VisualMelon** [comment]
The association should be broken when you change the `Model`/binding on the `PlotView`. In your case, it sounds like you have two separate documents, and you might want to consider using separate models for them: the restriction to one plot view per model is not arbitrary: the model (for better or worse) contains information needed for rendering and interacting with the plot.

If I'm understanding correctly,  the `IPlotModel.AttachPlotView` method may achieve what you want. Passing `null` as a parameter will remove the association with the plot view. This, however, won't inform the plot view that it has lost its model, so you'll have to juggle the views.

There is lots more discussion on this issue in https://github.com/oxyplot/oxyplot/issues/497

You are correct that `InvalidatePlot` is indeed about refreshing the view, and can't help you here: it will (optionally) update the data in series from their sources, and will ask the plot view to redraw.
